packages/ember-runtime/lib/mixins/array.js

```diff
@@ -1094,10 +1078,11 @@ const ArrayMixin = Mixin.create(Enumerable, {
     }
 
     let ret = A();
+    // SameValueZero comparison (NaN !== NaN)
+    let predicate = value === value ? item => item === value : item => item !== item;
```

Comment:
Why move this up to a function? Doesn’t seem to simplify or correct anything here...